app/scripts/controllers/containerfileviewer.js

```diff
@@ -66,6 +85,81 @@ angular.module('dockstore.ui')
           function(result){
             $scope.selTagName = $scope.successContent[0].tag;
             $scope.selDescriptorName = $scope.successContent[0].descriptor;
+            $scope.fileContent = $scope.successContent[0].content;
+            var result = $scope.fileContent.toLowerCase();
+            m = [];
+            v = false;
+            if($scope.selDescriptorName === "cwl"){
+              //Descriptor: CWL
+              if(result.search("inputs:") !== -1){
```

Comment:
This code seems pretty redundant. Could you break this out into a function that takes "inputs", "outputs", "baseCommand", etc. as a parameter?